lib/document/Document.js

```diff
@@ -57,6 +59,10 @@ class Document {
       delete data.$updatedAt;
     }
 
+    this.encodedProperties = this.dataContract.getEncodedProperties(
```

Comment:
Isn't it already cached in DataContract?

Reply:
Yep, it is